backend/services/useradm/utils/similar_chcker.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"strings"
+
+	"github.com/mendersoftware/mender-server/services/useradm/model"
+)
+
+func CheckIfPassSimilarToEmailRaw(mail string, pass string) bool {
+	if len(mail) == 0 || len(pass) == 0 {
+		return false
+	}
+	lmail := strings.ToLower(mail)
+	lpass := strings.ToLower(pass)
+	return strings.Contains(lmail, lpass)
+}
+func CheckIfPassSimilarToEmail(user *model.User, userUpdate *model.UserUpdate) bool {
+	if len(userUpdate.Password) > 0 {
+		// check when changing the password
+		mail := string(user.Email)
+		if len(mail) == 0 {
+			mail = string(userUpdate.Email)
+		}
+		return CheckIfPassSimilarToEmailRaw(mail, string(userUpdate.Password))
+	} else if len(userUpdate.Email) > 0 {
+		// check when changing the email
+		return CheckIfPassSimilarToEmailRaw(string(userUpdate.Email), string(userUpdate.CurrentPassword))
+	}
```

Comment:
You should only compare the password if it changed. Comparing the current password could possibly prevent you from ever changing the password if it happens to be bad.
```suggestion
	}
```

Reply:
By removing the "else if" check it will be possible to change your email to the password or similar to the password. If the password and email is already similar it shouldn't be a problem to changing them

Reply:
Very good observation!
So if the user changes the email, the email is similar to the current password that should also be prevented. However, we need to be careful to not compare the current password with the previous email, that would create the scenario I described above where a user cannot possibly change their password.